### ISSUE TITLE
Simple processor and Simple Log processor use a non-standard feature deprecated in C++20

### ISSUE THREAD

**Issue by maxgolov** [open-telemetry/opentelemetry-cpp#468]
These two pieces of code:

https://github.com/open-telemetry/opentelemetry-cpp/blob/68e510c68e2abff153fcc3cf02334fc24af92ee4/sdk/include/opentelemetry/sdk/trace/simple_processor.h#L75

https://github.com/open-telemetry/opentelemetry-cpp/blob/106c00a4a41d6aa00dac6ad19488efb8f1b4fbd5/sdk/include/opentelemetry/sdk/logs/simple_log_processor.h#L62

Should be rewritten as follows:

- assignment for anything that is below C++20, as shown here: https://en.cppreference.com/w/cpp/atomic/ATOMIC_FLAG_INIT
```cpp
std::atomic_flag static_flag = ATOMIC_FLAG_INIT; // static initialization,
```

- and empty constructor for C++20, since `This macro is no longer needed and deprecated, since default constructor of std::atomic_flag initializes it to clear state.`

I hit this while rebasing my PR for Standard Library support and testing it with C++20. The code won't compile as written in Visual Studio 2019 with C++20 enabled.

**Comment by maxgolov** [comment]
I'll fix it.

**Comment by maxgolov** [comment]
I added the fix in #374
